contracts/pallet-ibc/src/ics20/memo.rs

```diff
@@ -195,7 +195,7 @@ impl<T: Config + Send + Sync, S: Module + Clone + Default + PartialEq + Eq + Deb
 				Error::implementation_specific(format!("Failed to parse receiver account"))
 			})?
 			.into_account();
-		<T as Config>::HandleMemo::execute_memo(&packet_data, receiver).map_err(|e| {
+		<T as Config>::HandleMemo::execute_memo(packet, &packet_data, receiver).map_err(|e| {
```

Comment:
why don't we just send the entire packet, instead of the packet and the packet data?

Reply:
True.
pass packet and packet_data. to not deserialise two times.
or move all logic(deserialise and AccountIdConversion) from process_memo into <T as Config>::HandleMemo::execute_memo

Reply:
Done.